Userland/Applications/PixelPaint/GuideTool.cpp

```diff
@@ -138,4 +148,48 @@ void GuideTool::on_context_menu(Layer&, GUI::ContextMenuEvent& event)
     m_context_menu->popup(event.screen_position());
 }
 
+void GuideTool::on_keydown(GUI::KeyEvent& event)
+{
+    if (m_shift_held)
+        return;
+
+    if (event.key() == Key_Shift)
+        m_shift_held = true;
+}
+
+void GuideTool::on_keyup(GUI::KeyEvent& event)
+{
+    if (!m_shift_held)
+        return;
+
+    if (event.key() == Key_Shift) {
+        m_shift_held = false;
+    }
+}
+
+GUI::Widget* GuideTool::get_properties_widget()
+{
+    if (!m_properties_widget) {
+        m_properties_widget = GUI::Widget::construct();
+        m_properties_widget->set_layout<GUI::VerticalBoxLayout>();
+
+        auto& snapping_container = m_properties_widget->add<GUI::Widget>();
+        snapping_container.set_fixed_height(40);
+        snapping_container.set_layout<GUI::VerticalBoxLayout>();
+
+        auto& snapping_label = snapping_container.add<GUI::Label>("Hold Shift to snap:");
```

Comment:
Maybe we can try to shorten the label to "Snap offset" or similar? That way the widget would fit on the same line, keeping the same layout as the rest of the properties widgets. The key shortcut could be shown somewhere else, maybe in a tooltip?

Reply:
Great idea! I was wondering where to put the information to put Shift...